test/e2e/query_test.go

```diff
@@ -340,6 +348,37 @@ func TestQueryLabelValues(t *testing.T) {
 	)
 }
 
+func TestQueryWithAuthorizedSidecar(t *testing.T) {
+	t.Parallel()
+
+	s, err := e2e.NewScenario("e2e_test_query_authorized_sidecar")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, s))
+
+	prom, sidecar, err := e2ethanos.NewPrometheusAndSidecarWithBasicAuth(s.SharedDir(), s.NetworkName(), "alone", defaultPromConfig("prom-alone", 0, "", ""), defaultWebConfig(), e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	_ = s.StartAndWaitReady(prom, sidecar)
```

Comment:
Let's use `testutil.Ok(t, s.StartAndWaitReady(prom, sidecar))`

Reply:
I remember trying this earlier, but `s.StartAndWaitReady(prom, sidecar)` requires to have auth in it and it fails while checking  if prom is up or not